src/_backend_agg.cpp

```diff
@@ -212,8 +212,8 @@ agg::rect_i RendererAgg::get_content_extents()
 
     r.x1 = std::max(0, r.x1 - 1);
     r.y1 = std::max(0, r.y1 - 1);
-    r.x2 = std::max(r.x2 + 1, (int)width);
-    r.y2 = std::max(r.y2 + 1, (int)height);
+    r.x2 = std::min(r.x2 + 1, (int)width);
+    r.y2 = std::min(r.y2 + 1, (int)height);
```

Comment:
weird, I can't find when these lines were introduced. Git blame said they were last touched during the CXX refactor, but when I go to that commit, I can't find these lines anywhere...